lib/idx/remediators/GenericRemediator/GenericRemediator.ts

```diff
@@ -0,0 +1,79 @@
+import { IdxContext, NextStep, Input } from '../../types';
+import { Remediator } from '../Base/Remediator';
+import { unwrapFormValue, hasValidInputValue } from './util';
+import { proceed } from '../../proceed';
+
+export class GenericRemediator extends Remediator {
+  canRemediate(): boolean {
+    // only handle remediations that are able to submit form (xhr)
+    if (typeof this.remediation.action !== 'function') {
+      return false;
+    }
+
+    const inputs = this.getInputs();
+    const res = inputs.reduce((acc, input) => {
+      return acc && hasValidInputValue(input, this.values);
+    }, true);
+    return res;
+  }
+
+  getData() {
+    const data = this.getInputs().reduce((acc, { name }) => {
+      acc[name] = this.values[name];
+      return acc;
+    }, {});
+    return data;
+  }
+
+  getNextStep(_context?: IdxContext): NextStep {
+    const name = this.getName();
+    const inputs = this.getInputs();
+    
+    /* eslint-disable no-unused-vars, @typescript-eslint/no-unused-vars */
+    // excludes transformed fields
+    const { 
+      // http metas have been transformed to action
+      href, 
+      method, 
+      rel, 
+      accepts, 
+      produces, 
+      // value has been transform to inputs
+      value,
+      // will be transformed to a function that resolves IdxTransaction
+      action,
+      ...rest 
+    } = this.remediation;
+    /* eslint-enable no-unused-vars, @typescript-eslint/no-unused-vars */
+
+    // step to handle form submission
+    if (action) {
+      return { 
+        ...rest,
+        ...(!!inputs.length && { inputs }),
+        action: async (params?) => {
+          return proceed(this.authClient, {
```

Comment:
Is this the only line within the `Remediator` class (and derived classes) where `this.authClient` is consumed? If so, I think passing the `authClient` as a param to this function is a simpler than changing the constructor for all `Remediator` classes

Something like:
```javascript
getNextStep(authClient: OktaAuthInterface, _context?: IdxContext) {
 ...
}
```